src/AppBundle/Command/ImportSaleCommand.php

```diff
@@ -83,21 +81,13 @@ private function getSales($limit, $offset)
         return $response->json();
     }
 
-    private function saveSales($jsonSales, $existingInvoices)
+    private function saveSales($jsonSales, $repository)
     {
-        foreach ($jsonSales as $jsonSale) {
-            $exists = false;
-            foreach ($existingInvoices as $existing) {
-                if (
-                    $existing['invoice'] == $jsonSale['invoice'] &&
-                    $existing['licenseId'] == $jsonSale['licenseId'] &&
-                    $existing['pluginKey'] == $jsonSale['pluginKey']
-                ) {
-                    $exists = true;
-                }
-            }
+        $newCnt = 0;
 
-            if (!$exists) {
+        foreach ($jsonSales as $jsonSale) {
+            if ($repository->findIfSaleIsNew($jsonSale['invoice'], $jsonSale['licenseId'], $jsonSale['pluginKey'])) {
```

Comment:
You are referring to my change that stops importing sales if it doesn't find any new ones, right? Yes, if you delete non-recent sales, they won't be reimported. Is there any reason to delete old sales?

And I'm aware that sales sometimes get inserted out of order by AM, but they would have to be inserted as number 51 for this to be an issue. Or if there are any other new sales, there would have to be a gap from the last new sale of at least 50. The reason being that the first 50 sales are always imported, and if there are any new sales there, then the next 50 sales will always be imported, etc.

If this is an issue, then you are welcome to remove that bit if code, and I will remove it from my version as well once I stop running the server manually on my own machine. Importing sales is quite slow, so I would rather not wait 10-15 seconds for it each time.

Another option would be to add a --fast command line option for when importing sales.